cli/bpmetadata/tfconfig.go

```diff
@@ -0,0 +1,257 @@
+// with the blueprint
+func getBlueprintInterfaces(configPath string) *BlueprintInterface {
+	//load the configs from the dir path
+	mod, diags := tfconfig.LoadModule(configPath)
+	for _, diag := range diags {
+		if diag.Severity == tfconfig.DiagError {
+			return nil
+		}
+	}
+
+	var variables []BlueprintVariable
+	for _, val := range mod.Variables {
+		v := getBlueprintVariable(val)
+		variables = append(variables, v)
+	}
+
+	var outputs []BlueprintOutput
+	for _, val := range mod.Outputs {
+		o := getBlueprintOutput(val)
+
+		outputs = append(outputs, o)
+	}
+
+	i := &BlueprintInterface{
+		Variables: variables,
+		Outputs:   outputs,
+	}
+
+	return i
+}
+
+//build variable
+func getBlueprintVariable(modVar *tfconfig.Variable) BlueprintVariable {
+	return BlueprintVariable{
+		Name:        modVar.Name,
+		Description: modVar.Description,
+		Default:     modVar.Default,
+		Required:    modVar.Required,
+		VarType:     modVar.Type,
+	}
+}
+
+//build output
+func getBlueprintOutput(modOut *tfconfig.Output) BlueprintOutput {
+	return BlueprintOutput{
+		Name:        modOut.Name,
+		Description: modOut.Description,
+	}
+}
+
+// getBlueprintRequirements gets the variables and outputs associated
+// with the blueprint
+func getBlueprintRequirements(rolesConfigPath, servicesConfigPath string) BlueprintRequirements {
+	//parse blueprint roles
+	p := hclparse.NewParser()
+	rolesFile, _ := p.ParseHCLFile(rolesConfigPath)
+	r := parseBlueprintRoles(rolesFile)
+
+	//parse blueprint services
+	servicesFile, _ := p.ParseHCLFile(servicesConfigPath)
+	s := parseBlueprintServices(servicesFile)
+
+	return BlueprintRequirements{
+		Roles:    r,
+		Services: s,
+	}
+}
+
+// parseBlueprintRoles gets the roles required for the blueprint to be provisioned
+func parseBlueprintRoles(rolesFile *hcl.File) []BlueprintRoles {
+	var r []BlueprintRoles
+	iamContent, _, _ := rolesFile.Body.PartialContent(rootSchema)
+	for _, block := range iamContent.Blocks {
+		if block.Type == "locals" {
+			iamAttrs, _ := block.Body.JustAttributes()
+			for k, _ := range iamAttrs {
+				var iamRoles []string
+				attrValue, _ := iamAttrs[k].Expr.Value(nil)
+				ie := attrValue.ElementIterator()
+				for ie.Next() {
+					_, v := ie.Element()
+					iamRoles = append(iamRoles, v.AsString())
+				}
+
+				containerRoles := BlueprintRoles{
+					// TODO: (b/248123274) no good way to associate granularity yet
+					Granularity: "",
+					Roles:       iamRoles,
+				}
+
+				r = append(r, containerRoles)
+			}
+
+			// because we're only interested in the top-level locals block
+			break
+		}
+	}
+
+	return r
+}
+
+// parseBlueprintServices gets the gcp api services required for the blueprint
+// to be provisioned
+func parseBlueprintServices(servicesFile *hcl.File) []string {
+	var s []string
+	servicesContent, _, _ := servicesFile.Body.PartialContent(rootSchema)
+	for _, block := range servicesContent.Blocks {
+		if block.Type == "locals" {
```

Comment:
This is checking the `locals` block in the project factory instantiation such as [this](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/test/setup/iam.tf#L17). This is basically looking at the locals block in that file and getting all `apis`. We can further target it to just look for the `apis` attribute

Reply:
Gotcha, either is fine but we should standardize since I think GKE is an exception where we create multiple projects and most blueprints today do not have the apis in locals.

Reply:
I'll change this to target the project module's `activate_apis` attribute in a follow-up but we should standardize that moving forward.